### PR TITLE
Add option to show high alch values in loot tracker

### PR DESCRIPTION
**Pull request by Weol**
Add option to show GE or HA values in the loot tracker.

New config:
config

When both GE and HA is checked:
bothloot

When only GE is checked:
geloot

When only HA is checked:
haloot

### CODE REVIEW

**Comment by raiyni** [comment]
This should be an enum with ge, alch, or both

**Comment by Weol** [comment]
Config now looks like this:
config

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java [hunk 1]
```diff
@@ -45,17 +46,31 @@ default String getIgnoredItems()
...
 		name = "",
-		description = ""
+		description = "",
+		position = 2
 	)
 	void setIgnoredItems(String key);
 
+	@ConfigItem(
+		keyName = "valueType",
+		name = "Loot values",
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java]: `Price display`

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerBox.java [hunk 2]
```diff
@@ -303,12 +341,81 @@ private void buildItems()
...
+		if (quantity > 1)
+		{
+			return new StringBuilder()
+				.append(StackFormatter.quantityToStackSize(value))
+				.append(" gp (")
+				.append(StackFormatter.quantityToStackSize(value / quantity))
+				.append(" ea)")
+				.toString();
+		}
+		else
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerBox.java]: Having else here is unnecessary

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java [hunk 3]
```diff
@@ -45,17 +46,31 @@ default String getIgnoredItems()
...
+		description = "Save loot between client sessions (requires being logged in)",
+		position = 4
 	)
 	default boolean saveLoot()
 	{
 		return true;
 	}
-}
\ No newline at end of file
+
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java]: remove this empty line

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPlugin.java [hunk 4]
```diff
@@ -455,4 +466,10 @@ private LootTrackerItem buildLootTrackerItem(int itemId, int quantity)
 
 		return trackerRecords;
 	}
+
+	LootTrackerValueType getValueType()
+	{
+		return config.valueType();
+	}
+
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPlugin.java]: remove this empty line

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerValueType.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+	BOTH("Both");
+
+	private final String name;
+
+	@Override
+	public String toString()
+	{
+		return name;
+	}
+
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerValueType.java]: Remove this empty line

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPanel.java [hunk 6]
```diff
@@ -536,12 +553,38 @@ private void updateOverall()
...
 	private static String htmlLabel(String key, long value)
 	{
 		final String valueStr = StackFormatter.quantityToStackSize(value);
 		return String.format(HTML_LABEL_TEMPLATE, ColorUtil.toHexColor(ColorScheme.LIGHT_GRAY_COLOR), key, valueStr);
 	}
+
+	void setValueType(LootTrackerValueType type)
+	{
+		overallGeValueLabel.setVisible(
+			EnumSet.of(LootTrackerValueType.GRAND_EXCHANGE, LootTrackerValueType.BOTH).contains(type));
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPanel.java]: Just compare the enums directly and maybe abstract this logic. You have copy and pasted this a bit.

Comment (Weol) [reply runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPanel.java]: de0366819507da1b1d7aa15cc39f99176f5f9c62

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerValueType.java (hunk 5 above)
Comment (Weol) [reply runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerValueType.java]: de0366819507da1b1d7aa15cc39f99176f5f9c62

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPlugin.java (hunk 4 above)
Comment (Weol) [reply runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPlugin.java]: de0366819507da1b1d7aa15cc39f99176f5f9c62

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java (hunk 3 above)
Comment (Weol) [reply runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java]: de0366819507da1b1d7aa15cc39f99176f5f9c62

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerBox.java (hunk 2 above)
Comment (Weol) [reply runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerBox.java]: de0366819507da1b1d7aa15cc39f99176f5f9c62

**Comment by raiyni** [comment]
You can just click the resolved button. Github shows the commit history.

**Comment by deathbeam** [comment]
LootTrackerRecord.java 100644 → 100755 i dont think this is correct?

**Pull request closed without merging**